### PR TITLE
Upgrades `SensorBase` to support views

### PR DESCRIPTION
**Pull request by Mayankm96**
# Description

As a step towards supporting sensors as "Views", this PR upgrades the `SensorBase` class to follow a similar definition as `RobotBase` and aims to update the sensor implementations as well.

To-Dos:
- [x] Add CameraView class
- [ ] Add HeightScannerView class
- [ ] Add unit tests for the sensor view classes
- [x] Move all old sensor implementations to `compat` directory
- [x] Add warning for users to move to using sensor view classes

Fixes #3

## Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## Screenshots

Please attach before and after screenshots of the change if applicable.

## Checklist

- [ ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./orbit.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file

### CODE REVIEW

File: source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/camera/camera.py [hunk 1]
```diff
@@ -248,112 +233,135 @@ def set_world_pose_ros(self, pos: Sequence[float] = None, quat: Sequence[float]
...
-            if forward_dir == Gf.Vec3d(0, 0, 1):
-                up_axis = Gf.Vec3d(0, 1, 0)
-            elif forward_dir == Gf.Vec3d(0, 0, -1):
-                up_axis = Gf.Vec3d(0, -1, 0)
+        if not self._is_initialized:
+            raise RuntimeError("Camera is not initialized. Please call 'initialize(...)' first.")
+        # resolve indices
+        indices = self._backend_utils.resolve_indices(indices, self.count, self._device)
+        # create tensors for storing poses
+        positions = self._backend_utils.create_zeros_tensor((len(indices), 3), "float32", self._device)
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/camera/camera.py:288]: Shall we maybe commit to a specific backend in orbit? For example torch only?

File: source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/sensor_base.py [hunk 2]
```diff
@@ -32,32 +33,69 @@ def __init__(self, sensor_tick: float = 0.0):
...
+    def view(self) -> XFormPrimView:
+        """The underlying XFormPrimView of the sensor."""
+        return self._view
 
     @property
-    def timestamp(self) -> float:
-        """Simulation time of the measurement (in seconds)."""
-        return self._timestamp
+    def prim_paths(self) -> List[str]:
+        """The path to the camera prim."""
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/sensor_base.py:71]: camera?

File: source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/sensor_base.py [hunk 3]
```diff
@@ -75,36 +113,63 @@ def set_visibility(self, visible: bool):
...
+        # Set the initialized flag
+        self._is_initialized = True
 
-    def reset(self):
-        """Resets the sensor internals."""
-        # Set current time
-        self._timestamp = 0.0
-        # Set zero captures
-        self._frame = 0
+    def reset_buffers(self, sensor_ids: Optional[Sequence[int]] = None):
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/sensor_base.py:157]: Shouldn't this method not also called buffer(sensor_ids) to have the fresh data on reset?

Comment (Dhoeller19) [reply source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/sensor_base.py:157]: Also why did you add _buffer to the names? I feel like it was clearer before

**Comment by Mayankm96** [comment]
Closing this PR since we are handling it in the internal development branch. We are currently planning for a code release in mid-July with new sensor supports and other changes.

**Pull request closed without merging**